transport/grpc/outbound.go

```diff
@@ -107,7 +107,14 @@ func (o *Outbound) Call(ctx context.Context, request *transport.Request) (*trans
 
 	var responseBody []byte
 	var responseMD metadata.MD
-	invokeErr := o.invoke(ctx, request, &responseBody, &responseMD, start)
+	span, invokeErr := o.invoke(ctx, request, &responseBody, &responseMD, start)
+	// span.Finish() must be the last call made to any span instance
+	defer span.Finish()
+	// Service name match validation, return yarpcerrors.CodeInternal error if not match
+	if match, resSvcName := checkServiceMatch(request.Service, responseMD); !match {
+		// If service doesn't match => we got response => span must not be nil
+		return nil, transport.UpdateSpanWithErr(span, yarpcerrors.InternalErrorf("Service name does not match, expected: %s, Got: %s", request.Service, resSvcName))
```

Comment:
nit: let's break this line onto multiple for readability

Reply:
Resolved.